back-end/data_controller.rb

```diff
@@ -1,11 +1,33 @@
 class DataController < ApplicationController
-  def validate_params
+  TEMPERATURE = 'temperature'.freeze 
+  NOISE = 'noise'.freeze
+  PRESSURE = 'pressure'.freeze
+  HUMIDITY = 'humidity'.freeze
+  LIGHT = 'light'.freeze
+
+  VALID_METRICS = [TEMPERATURE, NOISE, PRESSURE, HUMIDITY, LIGHT].freeze
+
+  def valid_params?(metrics)
```

Comment:
You could simplify this to:

`metrics.all? { |metric| VALID_METRICS.include?(metric) }`